checks/system/integration/eiger.py

```diff
@@ -0,0 +1,225 @@
+    check('which vim    || echo FAILED', not_expected='FAILED')
+    check('which gcc    || echo FAILED', not_expected='FAILED')
+    check('which gcc-12 || echo FAILED', not_expected='FAILED')
+
+    check('which emacs || echo FAILED', not_expected='FAILED')
+
+#-----------------------------------------------------------------------------#
+#                                                                             #
+#                               Mount-points                                  #
+#                                                                             #
+#-----------------------------------------------------------------------------#
+
+    check.CLASS = 'MOUNTS'
+
+    check('grep -q "/users /users dvs"            /proc/mounts || echo FAILED', not_expected='FAILED')
+    check('grep -q "/store /store dvs"            /proc/mounts || echo FAILED', not_expected='FAILED')
+    check('grep -q "/project /project dvs"        /proc/mounts || echo FAILED', not_expected='FAILED')
+
+    check('grep -q "pe_opt_cray_pe /opt/cray/pe"  /proc/mounts || echo FAILED', not_expected='FAILED')
+    check('grep -q "pe_opt_AMD /opt/AMD"          /proc/mounts || echo FAILED', not_expected='FAILED')
+    check('grep -q "pe_opt_intel /opt/intel"      /proc/mounts || echo FAILED', not_expected='FAILED')
+
+    check('printenv SCRATCH || echo FAILED', not_expected='FAILED')
+    check('printenv PROJECT || echo FAILED', not_expected='FAILED')
+    check('printenv STORE   || echo FAILED', not_expected='FAILED')
+    check('printenv APPS    || echo FAILED', not_expected='FAILED')
+    check('printenv HOME    || echo FAILED', not_expected='FAILED')
+
+    check('bash -c "[[ $SCRATCH == /capstor/scratch/cscs/*  ]] || echo FAILED"', not_expected='FAILED')
+    check('bash -c "[[ $PROJECT == /project/*               ]] || echo FAILED"', not_expected='FAILED')
+    check('bash -c "[[ $STORE   == /store/*                 ]] || echo FAILED"', not_expected='FAILED')
+    check('bash -c "[[ $APPS    == /capstor/apps/cscs/eiger ]] || echo FAILED"', not_expected='FAILED')
+    check('bash -c "[[ $HOME    == /users/*                 ]] || echo FAILED"', not_expected='FAILED')
+
+    check('printenv TMP || echo FAILED', expected='FAILED')
+
+#-----------------------------------------------------------------------------#
+#                                                                             #
+#                                   Slurm                                     #
+#                                                                             #
+#-----------------------------------------------------------------------------#
+
+    check.CLASS = 'SLURM'
+
+    check('test -e /etc/slurm/slurm.conf || echo FAILED', not_expected='FAILED')
+    check('which sinfo || echo FAILED', not_expected='FAILED')
+    check('ps aux | grep munge', expected='/usr/sbin/munged')
+    check('scontrol ping', expected='Slurmctld\(primary\) at .* is UP')
```

Comment:
It's not a big deal, but maybe it's better to use raw string because I am getting some some syntax warning in some of the regex. For example:
```
 SyntaxWarning: invalid escape sequence '\('
```

Reply:
Where did you see this message?

Reply:
Hm, I am seeing that it may have to do with the version of python I am using (3.12.4). It appears when I running the test, at the loading stage:
```
...
/users/eirinik/repos/cscs-reframe-tests/checks/system/integration/eiger.py:179: SyntaxWarning: invalid escape sequence '\('
  check('scontrol ping', expected='Slurmctld\(backup\) at .* is UP')
[==========] Running 5 check(s)
[==========] Started on Tue Jul 30 14:28:20 2024+0200

[----------] start processing checks
...
```

Reply:
yep, it's fine now :)